templates/components/request-reset-view.jsx

```diff
@@ -45,8 +47,16 @@ var RequestResetPassword = React.createClass({
       </div>
     );
   },
+  handleTransition: function(page) {
+    this.transitionTo(page.to, '', page.query);
+  },
   handleFormData: function(data) {
-    this.props.submitForm(data);
+    if(!this.refs.userform.state.checkedUsername) {
```

Comment:
I thought that we are trying to avoid callback yesterday, no? Since we have done it with a callback before and it seems to be a bit hacky...